coiote-aws-iot-cloud-formation/certificates/certificates.py

```diff
@@ -0,0 +1,149 @@
+    send_internal_certificate(internal_certificate, internal_cert_ca)
+    save_internal_certificate_data(internal_certificate)
+    send_external_certificate(external_certificate)
+
+
+def generate_external_cert(email_address: str, common_name: str, serial_number=0) -> Certificate:
+    # can look at generated file using openssl:
+    # openssl x509 -inform pem -in selfsigned.crt -noout -text
+    # create a key pair
+    k = crypto.PKey()
+    k.generate_key(crypto.TYPE_RSA, 4096)
+    # create a self-signed cert
+    cert = crypto.X509()
+    cert.get_issuer().C = 'PL'
+    cert.get_issuer().ST = 'Lesser Poland'
+    cert.get_issuer().L = 'Cracow'
+    cert.get_issuer().O = 'AVSystem'
+    cert.get_issuer().OU = 'AVSystem'
+    cert.get_issuer().CN = 'AVSystem'
+    cert.get_subject().CN = common_name
+    cert.get_subject().emailAddress = email_address
+    cert.set_serial_number(serial_number)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(10 * 365 * 24 * 60 * 60)
+    cert.set_issuer(cert.get_subject())
+    cert.set_pubkey(k)
+    cert.sign(k, 'sha256')
+
+    return {
+        'certificatePem': crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode("utf-8"),
+        'privateKey': crypto.dump_privatekey(crypto.FILETYPE_PEM, k).decode("utf-8")
+    }
+
+
+def send_internal_certificate(internal_certificate, internal_cert_ca) -> Response:
+
+    request_body: InternalCertificate = {
+        'certificateAuthority': internal_cert_ca,
+        'certificatePem': internal_certificate['certificatePem'],
+        'privateKey': internal_certificate['keyPair']['PrivateKey']
+    }
+
+    uri = REST_URI + '/awsIntegration/auth/internalCertificate/' + GROUP_ID
+    return requests.post(uri, json=request_body, auth=(USER, PASSWORD))
+
+
+def save_internal_certificate_data(internal_certificate):
+    secrets_manager_client.create_secret(Name=cert_data_secret_name)
+    secret_string = json.dumps(internal_certificate['certificateId'])
+    secrets_manager_client.put_secret_value(SecretId=cert_data_secret_name, SecretString=secret_string)
+
+
+def send_external_certificate(external_certificate: Certificate) -> Response:
+    request_body: ExternalCertificate = {
+        'certificatePem': external_certificate['certificatePem']
+    }
+
+    uri = REST_URI + '/auth/certificates'
+    return requests.post(uri, json=request_body, auth=(USER, PASSWORD))
+
+
+def save_certificate_in_secret_manager(external_certificate: Certificate):
+    secrets_manager_client.create_secret(Name=cert_secret_name)
+    secrets_manager_client.put_secret_value(SecretId=cert_secret_name, SecretString=json.dumps(external_certificate))
+
+
+@helper.delete
+def delete(event, context):
+    try:
+        delete_internal_certificate_from_coiote()
+        delete_internal_certificate_from_aws()
+        delete_external_certificate()
+        delete_certificates_from_aws()
+    except Exception:
+        pass
```

Comment:
W sumie nie wiem czy chcemy tu łapać exceptiony, zgaduję, że jeśli będą złapane to nie zobaczymy w CloudWatchu co się zjepsuło